### PR TITLE
Zadanie trzecie

### PR DESCRIPTION
**Pull request by gorkowykamil**

### CODE REVIEW

**Comment by Patrycjadz0** [comment]
Zadanie wykonane poprawnie.

File: 02/ML_zadanie.ipynb
Comment (nowikarol) [inline 02/ML_zadanie.ipynb:1]: Wszystko w porządku. Nie mam uwag.

**Review by dwartalski — APPROVED** [review]
zadanie działa więc git , fajnie że zostawiłeś swoje stare próby

**Comment by jgilewicz** [comment]
Zadanie wykonane bardzo dobrze, pipeline uczenia wraz z modelem zachowany koncepcyjnie. Zadanie zaliczone. Wrzucam tylko kilka drobnych uwag:

- dla czytelności lepiej nie robić `from torchvision.transforms import functional as TF` bo wygląda jak import Tensorflow XD -wystarczy samo F
- staraj się raczej nie zostawiać magic numbers w kodzie `            nn.Conv2d(in_ch, out_ch, kernel_size=3, padding=1)` nawet jak nie ulegną zmianie, to w konstruktorze warto je zdefiniować jako bazowe parametry + z reguły ustawiamy zazwyczaj `padding = kernel_size // 2` 
- co do całego kodu sprawdzającego: albo warto sobie zrobić funkcję treningową, która przyjmuje parametry i odpalić to w pętli sprawdzającej, albo wykorzystać do tego taką bibliotekę jak [Optuna ](https://optuna.org) warto znać na przyszłość
- `visualize_predictions(model, val_ds, device)` wiem, że to tylko wizualizacja, ale jeśli chciałbyś sprawdzić accuracy modelu rzetelnie, to trzeba jeszcze wydzielić dataset testowy. Argumentem za może być to, ze na zbiorze walidacyjnym model nigdy się nie uczył, ale jest niepisana zasada dzielenia datasetu na  treningowy, walidacyjny i testowy (często w proporcjach 70:10:20)
- jeszcze komentarz odnośnie samych wyników (czysto informacyjny, bo zadanie trochę źle zostało sformułowane) - tak mały UNet nie poradzi sobie z segmentacją, potrzebne jest znaczące zwiększenie modelu + wydłużenie treningu na większym modelu.  Tutaj podsyłam do wglądu w [notatniku](https://github.com/keyurparalkar/Semantic-Segmentation-with-UNETs/blob/master/FASTai_UNET.ipynb) jak duży jest ten UNet zaimplementowany za pomocą [FastAI](https://github.com/fastai/fastai).
